src/Streamly/Internal/Data/Stream/Ahead.hs

```diff
@@ -53,6 +63,15 @@ import Prelude hiding (map)
 
 #include "Instances.hs"
 
+-- $setup
+-- >>> import Control.Concurrent (threadDelay)
+-- >>> :{
+--  delay n = do
+--      threadDelay (n * 1000000)   -- sleep for n seconds
+--      putStrLn (show n ++ " sec") -- print "n sec"
+--      return n                    -- IO Int
+-- :}
+
```

Comment:
Is this what you were talking about? doctests don't pick this up from the previous stuff (defined in the module header)?
I'm assuming `$setup` is identified by doctests?
Why did not put `import qualified Streamly.Prelude as Stream` here?

Reply:
The full thing is in module header, including this, to show in haddock docs for the user. setup section has only minimal required. Alternatively we can put everything in setup and use `$setup` in one export section named "Running Examples" or something. I did this in PR 10009.

Reply:
Of course, PR 10009.

Reply:
Not a big deal, but, if possible we should do it here as well. Works as a good standard. Reduces duplication and other errors surrounding it.